### ISSUE TITLE
SA1115: ParameterMustFollowComma

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#49]
http://www.stylecop.com/docs/SA1115.html

**Comment by macpak** [comment]
grabbed

**Comment by macpak** [comment]
I am in the process of implementing this one, and I am wondering about one thing:
* the description of this rule says:

> Cause

> A parameter within a C# method or indexer call or declaration does not begin on the same line as the previous parameter, or on the next line.

> Rule Description

> A violation of this rule occurs when there are one or more blank lines between a parameter and the previous parameter. 

So my understanding would be that for the example below, I will get just one violation, for parameter **s**

        public static void Fun(int i, int j,
            
            int s)
        {
        }

However, the original StyleCop reports 2 violations, the second one for parameter **j**. I think it's incorrect behaviour, my understanding would be that this rule should only check if there are blank lines between parameters. There is [SA1117 rule](http://stylecop.soyuz5.com/SA1117.html) which checks if all parameters are on one line or each one is located on a separate line.
If my understanding of this rule is incorrect, please let me know.

**Comment by sharwell** [comment]
Your understanding is correct. Only report one violation, even for the following:

```csharp
public static void Fun(int i, int j,

    int s, int t)
{
}
```

**Comment by macpak** [comment]
Thanks ! :+1:
